### ISSUE TITLE
[Feature Request] erweiterte Abfahrfüllstände

### ISSUE THREAD

**Issue by rohne83** [Stephan-S/FS19_AutoDrive#2097]
Hi,

1. wäre es machbar, den Abfahrfüllstand ähnlich wie bei CP mit Minimal- und Maximalwerten einzustellen? Derzeit ist es ja so, dass zB AD-Abfahrer im Modus Abholen+Abliefern beim Erreichen genau des eingestellten Wertes losfahren, unabhängig davon ob zB der Drescher oder Überladewagen von dem sie Beladen werden schon mit dem Überladen fertig ist oder nicht. Hilfreich wäre es ja, wenn er bei Erreichen des Min-Wertes trotzdem erstmal noch mit dem Abfahren wartet bis entweder das Überladen abgeschlossen oder der Max-Wert erreicht ist.

2. ebenfalls betreffend der Abfahrfüllstände hat man derzeit Probleme mit AD-Befüllkursen von durch CP-gesteuerte Sämaschinen, die einen recht kleinen Saatgut- und dafür einen ziemlich großen Düngertank haben (oft der Fall bei Einzelkornsämaschinen, zB der Vanilla-Horsch Maestro). Wenn die Sämaschine dann aber gar nicht Düngen sondern nur zB Mais legen soll, ist der Düngertank dann immer noch 100% voll wenn das Saatgut ausgegangen ist. Und egal was für einen Abfahrfüllstand ich einstelle, AD schickt den Fahrer immer direkt zum Feld zurück und nicht zum Silo Saatgut nachtanken. Wobei es für solche Geräte auch generell suboptimal ist, dass man das Nachfüllen des Düngertanks durch AD nicht komplett deaktivieren kann.

**Comment by Iwan1803** [comment]
zu 1.: das stimmt so nicht, denn der Abfahrer fährt nicht weg, bevor er nicht komplett abgetankt hat (z.Bsp. durch einen Drescher.) oder er voll beladen ist (z.Bsp. durch einen ÜLW).
Ist allerdings der eingestellte Füllstand erreicht und er wird nicht weiter befüllt, dann fährt er weg.

zu 2.: hier zählt der Füllstand von Dünger & Saatgut zusammen. Wenn die Saatmaschine "ungüstige" Verhältnisse zwischen beiden Tanks hat und man dann nicht auch düngt, kann sowas wie beschrieben passieren.

**Comment by rohne83** [comment]
zu 1. beim aktiven Drescher-Abfahr-Modus hast du Recht. Aber bei Abholen&Abliefern (also Abfahrer wartet am Feldrand auf Beladen durch selbstabtankenden Drescher oder ULW), haut der AD-Abfahrer immer genau dann ab sobald der eingestellte Füllstand erreicht ist und wartet nicht darauf dass der Überladevorgang abgeschlossen ist.

zu 2. Grade bei Lohnaufträgen oder je nachdem welche Düngestrategie man hat (die Giants-Engine lässt ja leider nicht viele Kombinationen zu bei denen es sich überhaupt lohnt Dünger gleichzeitig mit der Saat auszubringen) kommt man in die Situation leider ziemlich häufig, und das eben schon mit Vanilla-Fahrzeugen. Und unnötiges Düngen geht ja auch ziemlich ins Geld, Gibts da keine Einstellmöglichkeiten mit denen man sich da drum herum mogeln kann?

**Comment by 60erjonny** [comment]
Hallo

@rohne83  Zu 2:
Hoffe verstehe es jetzt nicht falsch...:-)
Da es sich hier wohl doch um Feldarbeit handelt , musst du das Problem mit CP angehen.
Hier kann man, in den Einstellungen die Düngefunktion abschalten.

**Comment by TobiHH78** [comment]
@60erjonny : Lies dir seinen Punkt 2 nochmal durch. Du hast das Problem scheinbar nicht verstanden...
Es geht nicht um die Feldarbeit an sich, sondern um das auffüllen durch AD.

**Comment by TobiHH78** [comment]
@60erjonny 
Nein, du hast es immer noch nicht verstanden, und dir den Eröffnungspost scheinbar überhaupt nicht durchgelesen...
Da steht:
**Wenn die Sämaschine dann aber gar nicht Düngen sondern nur zB Mais legen soll, ist der Düngertank dann immer noch 100% voll wenn das Saatgut ausgegangen ist. Und egal was für einen Abfahrfüllstand ich einstelle, AD schickt den Fahrer immer direkt zum Feld zurück und nicht zum Silo Saatgut nachtanken.**

Es geht darum, dass AD kein Saatgut auffüllt, wenn der Saatguttank bei einer Kombisämaschine leer ist, aber noch 100% Dünger vorhanden ist

**Comment by Iwan1803** [comment]
Immer ruhig bleiben 😉
Diesen Effekt habe ich übrigens noch mit keiner Saatmaschine gehabt - außer es war der falsche Modus eingestellt (Abholen& Abliefern statt nur Abholen).

@rohne83 
bitte mal poste mal einen Screenshot von den Abfahrereinstellungen und mit welcher Saatmaschine genau das passieren soll.

**Comment by rohne83** [comment]
@Iwan1803 zu 2. mein Fehler: ich nutze da zZ vA die Maestro aus dem Horsch-Pack mit 7000l Dünger- und 2000l Saatguttank. Hatte das "Auffüllen bis" auf einem recht niedrigen Wert in der Hoffnung damit irgendwie verhindern zu können, dass AD überhaupt auch den (zunächst leeren) Düngertank auffüllt. Wenn der aber eh schon voll ist, muss man natürlich den Wert auf mindestens 80% stellen, dann klappts auch mit dem Saatgut nachholen ;)

**Comment by Iwan1803** [comment]
7k zu 2k Liter ist aber auch ein sehr doofes Verhältnis, aber schön, wenn nun zumindest 2. geklärt ist 😉
Erledigt sich damit 1. auch gleich?

**Comment by rohne83** [comment]
Nein, der 1. sind ja die Abfahrer, und da ist das Verhalten ungewünschterweise immer wie beschrieben. Mal fix die typische Situation nachgestellt:
ULW (gesteuert von CP) hat 46000l Ladevolumen, wird fährt aber schon mit 90% zum Überladepunkt, weil es sich wegen ein paar tausend Litern nicht lohnt auf die nächste Drescherladung zu warten.
LKW (gesteuert von AD im Modus Abholen&Abliefern) hat 48000l Volumen, da der ULW teilweise schon mit ca 40000l ankommt ist sein Abfahrfüllstand auf 85% eingestellt (auch für ihn lohnt es sich nicht deswegen noch auf die nächste Fuhre zu warten). Sobald die 85% erreicht sind macht er mitten während des Überladevorgangs die Plane zu und fährt weg, obwohl im ULW (wenn doch mal voll) noch 5000l drin sind - gut zu sehen im zweiten Screenshot. Ohne automatische Abdeckungen kann man da zwar noch ein paar Liter zusätzlich ergaunern, aber das grundsätzliche Verhalten ließ sich so besser zeigen.
![fsScreen_2021_09_22_16_24_42](https://user-images.githubusercontent.com/71030138/134363664-ef2b9677-e72e-48ae-a519-160ffb044d3c.png)
![fsScreen_2021_09_22_16_25_59](https://user-images.githubusercontent.com/71030138/134363682-28afbaff-8798-488d-a138-de0ebd535a49.png)

**Comment by Iwan1803** [comment]
@rohne83 
danke für die ausführlichen Infos.

@Axel32019 
vielleicht solltest du dir das mal ansehen. Theoretisch sollte er den Anhänger ja komplett vollmachen, tut er aber nicht.
Ich schau mal, ob ich das bei mir nachstellen kann.

**Comment by Iwan1803** [comment]
Ist im aktuellen Release 1.1.1.8 gefixt:
https://github.com/Stephan-S/FS19_AutoDrive/releases/latest
